### PR TITLE
#215 Added CASystemMessageToJournal, improved journal commands, refactored SystemMessageHues

### PR DESCRIPTION
**Pull request by d0tsd0tsd0ts**
- [x] Added CASystemMessageToJournal
Now, CA SystemMessage will add in JournalEntry, so user will be able to look for text using InJournal that he wrote in SystemMessage or CA sent. 
I.E:
```
ClearJournal()
Pathfind( 647,2073,0 )
Pause(1000)
print InJournal("maximum")
print InJournal("distance","System")
print InJournal("distance","System",-1,1000)
```

- [x] Improved journal commands
Now, InJournal and WaitForJournal can use "alias" as author's param. InJournal has timeout as last param and that can be setted by user.

I.E:
```
SetAlias("beetle", 0x17ea42) #Pet Name = "a war horse"
ClearJournal()
Feed("beetle", 0x9d0,1, -1)
print InJournal("shies", "")
print InJournal("shies")
print InJournal("shies", "a war horse")
print InJournal("shies", "beetle")
print InJournal("shies", "beetle", -1, 300)
```

- [x] Refactored SystemMessageHues
I've put this on Enums.cs to sort...

### CODE REVIEW

File: ClassicAssist/UO/Data/Enums.cs [hunk 1]
```diff
@@ -157,4 +157,13 @@ public enum CharacterListFlags
         NewMovementSystem = 0x4000,
         NewFeluccaAreas = 0x8000
     }
+
+    [Flags]
```
Comment (Reetus) [inline ClassicAssist/UO/Data/Enums.cs:161]: Remove the [Flags], you can't specify multiple hues...

https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-6.0

**Review by Reetus — CHANGES_REQUESTED** [review]

File: ClassicAssist/UO/Data/Enums.cs (hunk 1 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/UO/Data/Enums.cs]: Ahh ok. I will remove. Ty so much ...

**Comment by d0tsd0tsd0ts** [comment]
Changed! Request is done! #215 branch

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs [hunk 2]
```diff
@@ -6,33 +6,44 @@
...
 using ClassicAssist.UO.Data;
 using ClassicAssist.UO.Network;
+using ClassicAssist.UO.Objects;
 
 namespace ClassicAssist.Data.Macros.Commands
 {
     public static class JournalCommands
     {
-        [CommandsDisplay( Category = nameof( Strings.Journal ),
-            Parameters = new[]
```
Comment (Reetus) [inline ClassicAssist/Data/Macros/Commands/JournalCommands.cs:15]: Put all these parameters back, they're used in the wiki docs

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs [hunk 3]
```diff
@@ -44,12 +55,8 @@ public static void ClearJournal()
             Engine.Journal.Clear();
         }
 
-        [CommandsDisplay( Category = nameof( Strings.Journal ),
-            Parameters = new[]
```
Comment (Reetus) [inline ClassicAssist/Data/Macros/Commands/JournalCommands.cs:48]: ^

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs [hunk 4]
```diff
@@ -93,11 +111,7 @@ void OnIncomingPacketHandlersOnJournalEntryAddedEvent( JournalEntry je )
         }
 
         // ReSharper disable once ExplicitCallerInfoArgument
-        [CommandsDisplay( "WaitForJournalArray", Category = nameof( Strings.Journal ),
-            Parameters = new[]
```
Comment (Reetus) [inline ClassicAssist/Data/Macros/Commands/JournalCommands.cs:97]: ^

**Review by Reetus — CHANGES_REQUESTED** [review]

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs (hunk 2 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/Data/Macros/Commands/JournalCommands.cs:15]: Done

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs (hunk 3 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/Data/Macros/Commands/JournalCommands.cs:48]: Done

File: ClassicAssist/Data/Macros/Commands/JournalCommands.cs (hunk 4 above)
Comment (d0tsd0tsd0ts) [reply ClassicAssist/Data/Macros/Commands/JournalCommands.cs]: Done

**Comment by d0tsd0tsd0ts** [comment]
Request is done!

**Pull request merged**